crates/types/src/dto.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::convert::Infallible;
+
+use restate_types_derive::BilrostNewType;
```

Comment:
Since this file is bilrost encoding related and along with the derive macro, we should create a separate crate for them. A common pattern is to have a crate with two sub-crates. The top level is the one that's exposed for users.

An example is in https://github.com/AhmedSoliman/prost-dto